providers/providerverifier.go

```diff
@@ -246,13 +247,20 @@ func (v *DefaultProviderVerifier) verifyCommitment(idt *oidc.Jwt, cic *clientins
 		if err != nil {
 			return fmt.Errorf("error marshalling jwk in cnf claim: %w", err)
 		}
-		cicJwkStr, err := json.Marshal(cic.PublicKey())
+		cicPublicKey, err := cic.PublicKey()
+		if err != nil {
+			return fmt.Errorf("error getting public key from CIC: %w", err)
+		}
+		cicJwkStr, err := json.Marshal(cicPublicKey)
 		if err != nil {
 			return fmt.Errorf("error marshalling jwk in CIC: %w", err)
 		}
 		if bytes.Equal(cicJwkStr, cnfJwkStr) {
 			return nil
 		}
+		// TODO: Is this the correct way to compare for equality?
+		// Given that we now store crypto.PublicKey and jwk.Key the JSON representation
+		// no longer matches.
```

Comment:
I was able to resolve the merge conflicts, but this part still fails the unit tests. 

My assumption is our implementation is too brittle for comparing keys, but I might be wrong. 

Do you plan on having the community meeting tomorrow @EthanHeilman ? Maybe we could take this opportunity to go over the PR and maybe get it over the finish line if there are no bigger comments to resolve.